### PR TITLE
fixed digi search behavior to check for end of list

### PR DESCRIPTION
**Pull request by hardenbr**

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
Just to make sure: this is a genuine correction - or does the occurence of "end" reveal a mistake elsewhere and this is just a fix?

File: HLTrigger/special/src/HLTRechitsToDigis.cc [hunk 1]
```diff
@@ -140,14 +140,17 @@ HLTRechitsToDigis::produce(edm::Event& iEvent, edm::EventSetup const& setup)  {
     // loop over the collection of rechits and match to digis
     EcalRecHitCollection::const_iterator ituneEB;
     for (ituneEB = recHitsHandle->begin(); ituneEB != recHitsHandle->end(); ituneEB++) {
-      EcalRecHit hit = (*ituneEB);
-      outputEBDigiCollection->push_back( (*digisEB->find(hit.id())).id(), (*digisEB->find(hit.id())).begin() );
+      EcalRecHit    hit = (*ituneEB);
```
Comment (fwyzard) [inline HLTrigger/special/src/HLTRechitsToDigis.cc]: could you change this to

`EcalRecHit const & hit = * ituneEB;`

to avoid the copy ?

**Comment by Martin-Grunewald** [comment]
please test

**Comment by hardenbr** [comment]
@Martin-Grunewald This is being looked into. At some point a rechit that does not have a corresponding DIGI is being created and passed to the RechitsToDigi module. This happens in a very small fraction of events which is why it wasn't found until it was put online.

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Martin-Grunewald** [comment]
please test

**Comment by hardenbr** [comment]
@Martin-Grunewald Does a PR need to be done for 80X?

**Comment by Martin-Grunewald** [comment]
Currently we have auto-port forward from 76 to 80...

**Comment by davidlt** [comment]
Add ref to issue: #11982

File: HLTrigger/special/src/HLTRechitsToDigis.cc [hunk 2]
```diff
@@ -140,14 +140,17 @@ HLTRechitsToDigis::produce(edm::Event& iEvent, edm::EventSetup const& setup)  {
     // loop over the collection of rechits and match to digis
     EcalRecHitCollection::const_iterator ituneEB;
     for (ituneEB = recHitsHandle->begin(); ituneEB != recHitsHandle->end(); ituneEB++) {
-      EcalRecHit hit = (*ituneEB);
-      outputEBDigiCollection->push_back( (*digisEB->find(hit.id())).id(), (*digisEB->find(hit.id())).begin() );
+      EcalRecHit const &    hit = (*ituneEB);
+      EcalDigiCollection::const_iterator digiLookUp = digisEB->find(hit.id());
+      // protect against a digi not existing
+      if( digiLookUp == digisEB->end()) continue;
```
Comment (shervin86) [inline HLTrigger/special/src/HLTRechitsToDigis.cc]: Reviewing this PR now.... this check makes no sense, there are no recHits without a digi associated. The recHit should not be skipped as it is now.

**Comment by shervin86** [comment]
@argiro @ferriff @emanueledimarco @ghezzi please have a look

**Comment by emanueledimarco** [comment]
I think that either the origin of the ghost hits is found (better) or the hits have to be skipped (patchy, but OK). Because this is what we will do anyway offline with them.

**Comment by ferriff** [comment]
RecHits with no digis might be there for recovered dead channels. Can this be checked comparing a list of DetId for which there is a recHit but no digi to the channel status in the DB?
However, from my fainted memories recoveries are switched off at HLT, and, if confirmed for Run2, I guess it would be better to fix a likely bug someplace else than integrate this fix...

**Pull request merged**